### ISSUE TITLE
Anonymized fields aren't being generated using HMA1

### ISSUE THREAD

**Issue by avico78** [sdv-dev/SDV#432]
### Environment details

* SDV version:
sdv==0.9.1
* Python version: 3.8

* Operating System: Linux/Docker

### Problem description
Using HMA1
I added an anonymize fields and i expect it will use it

### What I already tried

```python
    import sqlalchemy
    from sdv.relational import HMA1
#fetching data from DB
    employee = pd.read_sql_table('employee',con=synthetic_db_src_engine)
#fetching data from DB
    employee_details = pd.read_sql_table('employee_details',con=synthetic_db_src_engine)

#Decalring on anonym  fields
    annonym_fields = {"id" : {"type" : "id"
                                },
                      "name" : {"type" : "categorical",
                              "pii": True,
                              "pii_category": "name"
                                },
                      "birthdate" :
                               {"type": "categorical",
                                "pii": True,
                                "pii_category": "date.between('2015-01-01', '2015-01-05')"
                                },
                      "gender" :
                               {"type" : "categorical",
                                "pii": True,
                                 "pii_category": "gender"
                                }
     }
   
    metadata = Metadata()
   
    metadata.add_table(name='employee', data=employee,primary_key='id',fields = annonym_fields)
    metadata.add_table(name='employee_details', data=employee_details, parent='employee',foreign_key='employee_id', primary_key='id')

    model = HMA1(metadata)
    tables = {'employee': employee,'employee_details': employee_details,}
    
    model.fit(tables)
    new_data = model.sample(num_rows=10)


   # save new_data to table 
  
```

Actually annonym_fields - i tried several options to see if something can work 
i.e  "pii_category": "ssn"  but nothing , it doesn't impact on final sample (it takes values from original data).

**Comment by npatki** [comment]
I've tried to replicate your issue using the [demo HMA1](https://sdv.dev/SDV/user_guides/relational/index.html) test set and am not having any issues. Could you see if the code snippet below works for you? What I'm doing here is simply overriding the `country` column and turning into a random PII addresses.

```python
    from sdv import load_demo, Metadata
    from sdv.relational import HMA1

    tables = load_demo()
    metadata = Metadata()

    fields = { 
        'country': {
            'type': 'categorical',
            'transformer': 'label_encoding',
            'pii': True,
            'pii_category': 'address'
        }   
    }   
    
    metadata.add_table(name='users', data=tables['users'], primary_key='user_id', fields_metadata=fields)   
    metadata.add_table(name='sessions', data=tables['sessions'],
        primary_key='session_id', parent='users', foreign_key='user_id')   

    model = HMA1(metadata)
    model.fit(tables)
```

It's possible this is an issue with `sqlalchemy` or the way pandas is reading the tables. What I find interesting is that applying `date.between('2015-01-01', '2015-01-05')` or `gender` as categories is actually serving me an error: `raise ValueError('Category "{}" couldn\'t be found on faker'.format(category))`.

Not sure if this helps. Maybe someone else can chime in if they've also experienced this issue.

**Comment by csala** [comment]
> What I find interesting is that applying `date.between('2015-01-01', '2015-01-05')` or `gender` as categories is actually serving me an error: `raise ValueError('Category "{}" couldn\'t be found on faker'.format(category))`.

This one is actually expected. The `pii_category` can currently only be one of the valid [Faker provider categories](https://faker.readthedocs.io/en/master/providers.html). This is actually specified on the section that talks about how to anonymize PII in the [Single Table User Guides](https://sdv.dev/SDV/user_guides/single_table/gaussian_copula.html#anonymizing-personally-identifiable-information-pii), but maybe it should also be added to the [Relational Metadata User Guide](https://sdv.dev/SDV/user_guides/relational/relational_metadata.html) too.

**Comment by npatki** [comment]
Yes, meant was that the error is expected, however @avico78 didn't mention that there was an error thrown, just that it was silently creating incorrect data.

Now that I'm looking at the code again, is it possible the error is due to the kwargs that are passed in?

@avico78 's example uses `fields=` kwarg

However, mine was working with `fields_metadata=`

**Comment by avico78** [comment]
@csala @npatki  thanks for checking , is that means only one field can be using the fake ?

**Comment by npatki** [comment]
You should be able to anonymize (fake) as many columns as you need by adding it to the dict.
 
Eg.

```python
anonym_fields = { 
    'address': {
        'type': 'categorical',
        'pii': True,
        'pii_category': 'address' 
    } , 'social': {
        'type': 'categorical',
        'pii': True,
        'pii_category': 'ssn'
   }}   
```
I'd try passing in dict using the `fields_metadata` kwarg, not `fields`.

```python
metadata.add_table(name='employee', data=employee, primary_key='id', fields_metadata=annonym_fields)

# NOT  fields=anonym_fields
```

**Comment by avico78** [comment]
Thanks @npatki

**Comment by npatki** [comment]
That's because `metadata_fields` is not inside the json; it's a kwarg you use when creating the `Metadata` object. See [user guide](https://sdv.dev/SDV/user_guides/relational/relational_metadata.html)

Specifically, the `add_table` call is a function that takes in arguments, the metadata json being one of those arguments. See also the [API](https://sdv.dev/SDV/api_reference/metadata/api/sdv.metadata.dataset.Metadata.add_table.html#sdv.metadata.dataset.Metadata.add_table).

Let me know if that helps!
